crates/ruff_python_formatter/src/expression/expr_slice.rs

```diff
@@ -91,7 +91,7 @@ impl FormatNodeRule<ExprSlice> for FormatExprSlice {
         if !all_simple && lower.is_some() {
             space().fmt(f)?;
         }
-        token(":").fmt(f)?;
+        write!(f, [soft_line_break(), token(":")])?;
```

Comment:
We should move this into the `else` branch or we may end up with an unintentional a `\n<space>`